tests/lax_numpy_test.py

```diff
@@ -2303,6 +2302,44 @@ def testFull(self, shape, fill_value_dtype, out_dtype, rng_factory):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+       {"testcase_name":
+           "_shape={}_n={}_axis={}_prepend={}_append={}".format(
+           jtu.format_shape_dtype_string(shape, dtype),
+           n, axis, prepend, append),
+        "shape": shape, "dtype": dtype, "n": n, "axis": axis,
+        "prepend": prepend, "append": append,
+        "rng_factory": jtu.rand_default}
+       for shape, dtype in _shape_and_dtypes(nonempty_nonscalar_array_shapes, default_dtypes)
+       for n in [0, 1, 2]
+       for axis in list(range(-len(shape), max(1, len(shape))))
+       for prepend in [None, 1, np.zeros(shape, dtype=dtype)]
+       for append in [None, 1, np.zeros(shape, dtype=dtype)]
+       ))
+  def testDiff(self, shape, dtype, n, axis, prepend, append, rng_factory):
+    rng = rng_factory(self.rng())
+    args_maker = lambda: [rng(shape, dtype)]
+
+    def np_fun(x, n=n, axis=axis, prepend=prepend, append=append):
+      if prepend is None:
+          prepend = np._NoValue
+      elif not np.isscalar(prepend) and prepend.dtype == jnp.bfloat16:
+          prepend = prepend.astype(np.float32)
+
+      if append is None:
+        append = np._NoValue
+      elif not np.isscalar(append) and append.dtype == jnp.bfloat16:
+        append = append.astype(np.float32)
+
+      if x.dtype == jnp.bfloat16:
+        return np.diff(x.astype(np.float32), n=n, axis=axis, prepend=prepend, append=append).astype(jnp.bfloat16)
+      else:
+        return np.diff(x, n=n, axis=axis, prepend=prepend, append=append)
+
+    jnp_fun = lambda x: jnp.diff(x, n=n, axis=axis, prepend=prepend, append=append)
+    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=(dtype != np.float16))
```

Comment:
Running a fuller set of tests revealed some more dtype issues here (for example, numpy tends to promote things aggressively to float64). The easiest thing would probably be to set `check_dtypes=False`

Reply:
Ok. I've added `check_dtypes=False`.